### PR TITLE
Fix pyarrow CI

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Got

```
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
```

on pyarrow CI (https://github.com/apache/arrow-datafusion/actions/runs/5972076068/job/16202226846?pr=7404). This patch tries to fix it.

### CODE REVIEW

**Comment by viirya** [comment]
`cargo test pyarrow` passes now.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -363,9 +363,24 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: "3.8"
+      - name: Install OpenSSL 1.1.1
```
Comment (tustvold) [inline .github/workflows/rust.yml:366]: Is there no apt package for this?

**Comment by tustvold** [comment]
Trying out a potentially simpler fix in https://github.com/apache/arrow-datafusion/pull/7416 :crossed_fingers:

File: .github/workflows/rust.yml (hunk 1 above)
Comment (viirya) [reply .github/workflows/rust.yml:366]: The images is installed with OpenSSL3.

Comment (viirya) [reply .github/workflows/rust.yml:366]: But the installed Python3.8 requires OpenSSL 1.1.1 in particular.

Comment (viirya) [reply .github/workflows/rust.yml:366]: We either install OpenSSL 1.1.1 like this PR, or use Python which uses OpenSSL 3, or use an image installed with OpenSSL 1.1.1.

Comment (tustvold) [reply .github/workflows/rust.yml:366]: Indeed https://github.com/apache/arrow-datafusion/pull/7416 simply continues to use the bullseye rust image, and I have filed an upstream bug report for them to fix the action to not rely on an EOL OpenSSL version.

**Comment by alamb** [comment]
This one was simpler: https://github.com/apache/arrow-datafusion/pull/7416 - which I think went with @viirya 's suggestion to use an image with preinstalled OpenSSL 3

**Comment by viirya** [comment]
Yea, sounds good to me. I don't know which to image to use last night. 😂

**Pull request closed without merging**